### PR TITLE
Dev/restructure

### PR DESCRIPTION
**Pull request by SciathGH**

### CODE REVIEW

File: server/src/main/kotlin/net/starlegacy/feature/progression/ShipKillXP.kt [hunk 1]
```diff
@@ -35,7 +36,7 @@ import org.bukkit.event.EventHandler
 import org.bukkit.event.EventPriority
 import org.bukkit.event.entity.PlayerDeathEvent
 import java.time.Instant
-import java.util.UUID
+import java.util.*
```
Comment (Peter-Crawley) [inline server/src/main/kotlin/net/starlegacy/feature/progression/ShipKillXP.kt:39]: No wildcard imports

File: server/src/main/kotlin/net/starlegacy/feature/progression/Levels.kt [hunk 2]
```diff
@@ -16,7 +17,7 @@ import net.starlegacy.util.msg
 import net.starlegacy.util.title
 import org.bukkit.Bukkit
 import org.bukkit.entity.Player
-import java.util.UUID
+import java.util.*
```
Comment (Peter-Crawley) [inline server/src/main/kotlin/net/starlegacy/feature/progression/Levels.kt:20]: No wildcard imports

**Review by Peter-Crawley — CHANGES_REQUESTED** [review]

File: server/src/main/kotlin/net/starlegacy/feature/progression/Levels.kt (hunk 2 above)
Comment (SciathGH) [reply server/src/main/kotlin/net/starlegacy/feature/progression/Levels.kt:20]: I hate intellij

File: server/src/main/kotlin/net/horizonsend/ion/server/misc/Audience.kt [hunk 3]
```diff
@@ -1,19 +1,24 @@
...
-}
+private val Audience.prefix
+	get() = when (this) {
+		is Player -> "to $name: "
+		else -> ""
+	}
 
 /** Sends a message to the audience with the given colour. */
-private fun Audience.sendColouredMessage(message: String, color: Int) = sendMessage(text(message, color(color)))
+private fun Audience.sendColouredMessage(message: String, color: Int) = sendMessage(
```
Comment (Peter-Crawley) [inline server/src/main/kotlin/net/horizonsend/ion/server/misc/Audience.kt:17]: No

File: server/src/main/kotlin/net/horizonsend/ion/server/registrations/configs/ServerConfiguration.kt [hunk 4]
```diff
@@ -9,7 +9,7 @@ import org.spongepowered.configurate.objectmapping.ConfigSerializable
 @ConfigSerializable
 data class ServerConfiguration(
 	val serverName: String? = null,
-	val ParticleColourChoosingMoneyRequirement: Double? = 5.0,
+	val particleColourChoosingMoneyRequirement: Double? = 5.0,
```
Comment (Peter-Crawley) [inline server/src/main/kotlin/net/horizonsend/ion/server/registrations/configs/ServerConfiguration.kt:12]: remove this

**Review by Peter-Crawley — CHANGES_REQUESTED** [review]
a few package changes i want:
`nhi.server.features.*` -> `nhi.server`
`nhi.server.misc.registrations.*` -> `nhi.server`
`nhi.server.misc.registration.configs` -> `nhi.server.config`
`nhi.server.misc.registration.Keys` -> `nhi.server.NamespacedKeys`

File: server/src/main/kotlin/net/horizonsend/ion/server/misc/Audience.kt (hunk 3 above)
Comment (SciathGH) [reply server/src/main/kotlin/net/horizonsend/ion/server/misc/Audience.kt:17]: bro this is spotless, if you this fixed, fix spotless

Comment (Peter-Crawley) [reply server/src/main/kotlin/net/horizonsend/ion/server/misc/Audience.kt:17]: *sorry, meant the lines below, you changed it to use minimessage*

Comment (SciathGH) [reply server/src/main/kotlin/net/horizonsend/ion/server/misc/Audience.kt:17]: why shouldnt it use minimessage? Why are we switching all the sudden

**Review by Peter-Crawley — APPROVED** [review]

**Pull request merged**